src/RoomDirectoryModel.cpp

```diff
@@ -0,0 +1,186 @@
+// SPDX-FileCopyrightText: 2021 Nheko Contributors
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "RoomDirectoryModel.h"
+#include "Cache.h"
+#include "ChatPage.h"
+
+#include <algorithm>
+
+RoomDirectoryModel::RoomDirectoryModel(QObject *parent, const std::string &s)
+  : QAbstractListModel(parent)
+  , server_(s)
+  , canFetchMore_(true)
+{
+        connect(this,
+                &RoomDirectoryModel::fetchedRoomsBatch,
+                this,
+                &RoomDirectoryModel::displayRooms,
+                Qt::QueuedConnection);
+}
+
+QHash<int, QByteArray>
+RoomDirectoryModel::roleNames() const
+{
+        return {
+          {Roles::Name, "name"},
+          {Roles::Id, "roomid"},
+          {Roles::AvatarUrl, "avatarUrl"},
+          {Roles::Topic, "topic"},
+          {Roles::MemberCount, "numMembers"},
+          {Roles::Previewable, "canPreview"},
+        };
+}
+
+void
+RoomDirectoryModel::resetDisplayedData()
+{
+        beginResetModel();
+
+        prevBatch_    = "";
+        nextBatch_    = "";
+        canFetchMore_ = true;
+
+        beginRemoveRows(QModelIndex(), 0, static_cast<int>(publicRoomsData_.size()));
+        publicRoomsData_.clear();
+        endRemoveRows();

```

Comment:
I don't think the remove rows calls are necessary if you are resetting the model

Reply:
@deepbluev7 Can you confirm this?

Reply:
I can and I told you that before.